packages/flutter/lib/src/widgets/bottom_navigation_bar_item.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/foundation.dart';
+
+import 'framework.dart';
+
+/// An interactive button within either material's [BottomNavigationBar]
+/// or the iOS themed [CupertinoTabBar] with an icon and title.
+///
+/// Should not be used on its own but embedded in one of the bottom navigation
+/// widgets above.
+///
+/// See also:
+///
+///  * [BottomNavigationBar]
+///  * <https://material.google.com/components/bottom-navigation.html>
+///  * [CupertinoTabBar]
+///  * <https://developer.apple.com/ios/human-interface-guidelines/ui-bars/tab-bars>
+class BottomNavigationBarItem {
+  /// Creates an item that is used with [BottomNavigationBar.items].
+  ///
+  /// The arguments [icon] and [title] should not be null.
+  BottomNavigationBarItem({
+    @required this.icon,
+    @required this.title,
+    this.backgroundColor
+  }) {
+    assert(icon != null);
+    assert(title != null);
+  }
+
+  /// The icon of the item.
+  ///
+  /// Typically the icon is an [Icon] or an [ImageIcon] widget. If another type
+  /// of widget is provided then it should configure itself to match the current
+  /// [IconTheme] size and color.
+  final Widget icon;
+
+  /// The title of the item.
+  final Widget title;
+
+  /// The color of the background radial animation for material [BottomNavigationBar].
+  ///
+  /// If the navigation bar's type is [BottomNavigationBarType.shifting], then
+  /// the entire bar is flooded with the [backgroundColor] when this item is
+  /// tapped.
+  ///
+  /// Not used for [CupertinoTabBar]. Control the invariant bar color directly
+  /// via [CupertinoTabBar.backgroundColor].
+  ///
+  /// See also:
+  ///  * [Icon.backgroundColor] and [ImageIcon.backgroundColor] to control the foreground color of
```

Comment:
I'm hoping this comment is wrong, because otherwise we really picked some bad names for those fields...